golang/pkg/dagent/utils/environment.go

```diff
@@ -0,0 +1,85 @@
+package utils
+
+import (
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+
+	"github.com/joho/godotenv"
+)
+
+const dirPerm = 0o700
```

Comment:
600 isn't sufficient enough?

Reply:
If it would be file, then yes, but this is for directori(es).
7 is RWX, so
R: read contents
W: write into it
X: open / cd into it
I think we need all of them.
Random note: Linux defaults to 755 for directories 644 for files.